projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -1028,85 +1983,205 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
+     * const unpinnedColumns = this.grid.unpinnedColumns.
+     * ```
+     */
     get unpinnedColumns(): IgxColumnComponent[] {
         return this._unpinnedColumns.filter((col) => !col.hidden); // .sort((col1, col2) => col1.index - col2.index);
     }
 
+    /**
+     * Returns the `IgxColumnComponent` that matches name.
+     * ```typescript
+     * const myCol = this.grid1.getColumnByName("ID");
+     * ```
+     * @param name
+     */
     public getColumnByName(name: string): IgxColumnComponent {
         return this.columnList.find((col) => col.field === name);
     }
 
+    /**
+     * Returns the `IgxColumnComponent` that matches index.
+     * ```typescript
+     * const myRow = this.grid1.getRowByIndex(1);
+     * ```
+     * @param index
+     */
     public getRowByIndex(index: number): IgxGridRowComponent {
         return this.gridAPI.get_row_by_index(this.id, index);
     }
 
+    /**
+     * Returns `IgxGridRowComponent` only if primary key is specified in
+     * the `IgxGridComponent` and the specified keyValue exist as a rowID in the `IgxGridComponent`.
+     * ```typescript
+     * const myRow = this.grid1.getRowByKey("cell5");
+     * ```
+     * @param keyValue
+     */
     public getRowByKey(keyValue: any): IgxGridRowComponent {
         return this.gridAPI.get_row_by_key(this.id, keyValue);
     }
 
+    /**
+     * Returns an array of visbile `IgxColumnComponent`s.
+     * ```typescript
+     * const visibleColumns = this.grid.visibleColumns.
+     * ```
+     */
     get visibleColumns(): IgxColumnComponent[] {
         return this.columnList.filter((col) => !col.hidden);
     }
 
+    /**
+     * Returns the `IgxGridCellComponent` that matches the conditions.
+     * ```typescript
+     * const myCell = this.grid1.getCellByColumn(2,"UnitPrice");
+     * ```
+     * @param rowIndex
+     * @param columnField
+     */
     public getCellByColumn(rowIndex: number, columnField: string): IgxGridCellComponent {
         const columnId = this.columnList.map((column) => column.field).indexOf(columnField);
         if (columnId !== -1) {
             return this.gridAPI.get_cell_by_index(this.id, rowIndex, columnId);
         }
     }
 
+    /**
+     * Returns `IgxGridCellComponent` only if primary key is specified in the grid.
+     * ```typescript
+     * grid.getCellByKey(1, 'index');
+     * ```
+     * @param rowSelector match any rowID
+     * @param columnField
+     */
     public getCellByKey(rowSelector: any, columnField: string): IgxGridCellComponent {
         return this.gridAPI.get_cell_by_key(this.id, rowSelector, columnField);
     }
 
+    /**
+     * Returns the total number of pages.
+     * ```typescript
+     * const totalPages = this.grid.totalPages;
+     * ```
+     */
     get totalPages(): number {
         if (this.pagingState) {
             return this.pagingState.metadata.countPages;
         }
         return -1;
     }
 
+    /**
+     * Returns the total number of records.
```

Comment:
as it is returned by the paging state this might only be available if paging is enabled - please double check